### ISSUE TITLE
Match reason function has access violations

### ISSUE THREAD

**Issue by Baltoli** [runtimeverification/llvm-backend#700]
While working on LLDB support, I found that the `match_N` / `LABEL.match` functions segfault on valid terms. This error doesn't actually seem to affect the results:
```
(lldb) expr resetMatchReason()
(lldb) expr getMatchLogSize()
(size_t) $44 = 0
(lldb) expr ((void (*)(block *))(match_103))(subject)
error: Execution was interrupted, reason: EXC_BAD_ACCESS (code=2, address=0x7ff7bfefefa0).
The process has been returned to the state before expression evaluation.
(lldb) expr getMatchLogSize()
(size_t) $45 = 1
(lldb) expr getMatchLog()[0]
(MatchLog) $46 = {
  kind = SUCCESS
  function = 0x0000000000000000
  debugName = 0x0000000000000000
  result = 0x0000000000000000
  args = size=0 {}
  pattern = 0x0000000000000000
  subject = 0x0000000000000000
  sort = 0x0000000000000000
}
```

Because I'm already in the debugger using the expression evaluator here, I can't drill down further into why this is happening.

To reproduce this, I think the best thing to do would be to build a semantics that includes a C++ hook that explicitly calls the relevant match checking function, and debug from there. Would need to grab the entire config as an argument using this trick:
```k
  syntax K ::= #dumpConfig() [function]
             | #dumpConfig(GeneratedTopCell) [function]

  rule [[ #dumpConfig() => #dumpConfig(<generatedTop> TC </generatedTop>) ]]
    <generatedTop> TC </generatedTop>
  rule #dumpConfig(GTC) => #log(#unparseKORE(GTC))
```

**Comment by Baltoli** [comment]
Can be reinvestigated now that `k-rule-apply` is merged.

**Comment by Robertorosmaninho** [comment]
This issue was already fixed with `k-rule-apply` PR #721.
